app/models/user.rb

```diff
@@ -87,6 +87,25 @@ def github_import=(import)
     @github_import = import
   end
 
+  %i(opted_in_newsletter opted_in_announcements opted_in_marketing_announcements opted_in_surveys opted_in_sponsorships opted_in_applications_open).each do |attribute|
+    define_method attribute.to_sym do
+      send("#{attribute}_at").present?
+    end
+
+    define_method "#{attribute}?".to_sym do
```

Comment:
From what I remember, this is fine to be a string, no need for a `to_sym` (same for the other methods).